>From 7e49ff036892fade86c7734118ee0c6da5cbad09 Mon Sep 17 00:00:00 2001
From: Hans de Goede <hdegoede@redhat.com>
Date: Thu, 11 Aug 2011 12:11:16 +0200
Subject: [PATCH 28/28] usb-redir: Don't try to write to the chardev after a
 close event

Sicne we handle close async in a bh, do_write and thus write can get
called after receiving a close event. This patch adds a check to
the usb-redir write callback to not do a qemu_chr_write on a closed
chardev.

Signed-off-by: Hans de Goede <hdegoede@redhat.com>
---
 usb-redir.c |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/usb-redir.c b/usb-redir.c
index 6d8f986..732ddab 100644
--- a/usb-redir.c
+++ b/usb-redir.c
@@ -226,7 +226,7 @@ static int usbredir_write(void *priv, uint8_t *data, int count)
     USBRedirDevice *dev = priv;
     int r;
 
-    if (dev->cs->write_blocked) {
+    if (!dev->cs->opened || dev->cs->write_blocked) {
         return 0;
     }
 
-- 
1.7.5.1

